historical_surveillance/views.py

```diff
@@ -0,0 +1,328 @@
+                                         category_of_school=school_category).exists():
+                form.save()
+                return redirect("/")
+
+            else:
+                error_message = "The record you are trying to create already exists."
+
+        stu = {
+            "name_of_school": data,
+            "error_message": error_message,
+            "form" : form,
+        }
+        return render(request, 'school.html', stu)
+    return render(request, 'school.html', {'form':form, 'name_of_school':data})
+
+
+# This function controls the enrollment by school,
+# grade and gender form view, processes the and submits form data to the database
+def enrollment(request):
+    form = EnrollmentForms()
+    district_df = pd.DataFrame(District.objects.all().values())
+    districts_names = district_df['District_Name']
+    return render(request, 'enrollment.html', {'form': form, 'district_names': districts_names})
+
+
+# This function controls the total enrollment and school capacity form, processes the form data and submits
+# it to the database
+def enroll_class(request):
+    form = AggregateEnrollmentForms()
+    if request.method == 'POST':
+        form = AggregateEnrollmentForms(request.POST)
+        if form.is_valid():
+            school_form = form.cleaned_data['name_of_school']
+            academic_year_form = form.cleaned_data['academic_year']
+            district_form = form.cleaned_data['district_of_school']
+            if not AggregateEnrollment.objects.filter(name_of_school=school_form, academic_year=academic_year_form,
+                                                      district_of_school=district_form).exists():
+                form.save()
+                return redirect("/")
+        else:
+            messages.add_message(request, messages.INFO, 'Data has been entered previously')
+
+    return render(request, 'enroll_class.html', {'form': form})
+
+
+# This view controls the enrollment / Capacity views table
+def enrolled(request):
+    data = AggregateEnrollment.objects.all()
+
+    stu = {
+        "name_of_school": data
+    }
+    return render(request, 'enrolled.html', stu)
+
+#change this to get a form to select the district and pass it as a parameter to present filter the table and present
+# the data for each district
+def enrolled_trend_district_1(request):
+    error_message = None
+    district_df = pd.DataFrame(District.objects.all().values())
+    districts_names = district_df['District_Name']
+    if district_df.shape[0] > 0:
+        if request.method == 'POST':
+            district_selected = request.POST.get('district_name', False)
+            if not district_selected:
+                error_message = "Please select a chart type and school"
+            else:
+                if district_selected is not False:
+                    data = AggregateEnrollment.objects.all().filter(district_of_school=district_selected)
```

Comment:
remove `.all()`

`AggregateEnrollment.objects.filter(district_of_school=district_selected)`

Reply:
Also if you set this up to use a foreign key. or. a one. to one you. can actually. do a reverse binding. Works pretty cool. 

`data = District.objects.get(code= district_selected).aggregate_enrollment_set`

Basically if you can query for the district it will check hey which aggregate enrollment objects have. a foreign key to the district and provide an easy way to access them.

Heres a cool example: https://stackoverflow.com/questions/37304816/django-one-to-many-relationship-and-reverse-lookup